stan/math/opencl/kernel_generator/operation_cl_lhs.hpp

```diff
@@ -38,32 +42,72 @@ class operation_cl_lhs : public operation_cl<Derived, Scalar, Args...> {
   inline kernel_parts get_kernel_parts_lhs(
       std::set<const operation_cl_base*>& generated, name_generator& name_gen,
       const std::string& i, const std::string& j) const {
+    if (generated.count(this) == 0) {
+      this->var_name = name_gen.generate();
+    }
+    std::string i_arg = i;
+    std::string j_arg = j;
+    this->derived().modify_argument_indices(i_arg, j_arg);
     std::array<kernel_parts, N> args_parts = index_apply<N>([&](auto... Is) {
       return std::array<kernel_parts, N>{
-          std::get<Is>(arguments_)
-              .get_kernel_parts_lhs(generated, name_gen, i, j)...};
+          std::get<Is>(this->arguments_)
+              .get_kernel_parts_lhs(generated, name_gen, i_arg, j_arg)...};
     });
     kernel_parts res{};
     res.body = std::accumulate(
         args_parts.begin(), args_parts.end(), std::string(),
         [](const std::string& a, const kernel_parts& b) { return a + b.body; });
     if (generated.count(this) == 0) {
       generated.insert(this);
-      this->var_name = name_gen.generate();
       res.args
           = std::accumulate(args_parts.begin(), args_parts.end(), std::string(),
                             [](const std::string& a, const kernel_parts& b) {
                               return a + b.args;
                             });
       kernel_parts my_part = index_apply<N>([&](auto... Is) {
         return this->derived().generate_lhs(
-            i, j, std::get<Is>(arguments_).var_name...);
+            i, j, std::get<Is>(this->arguments_).var_name...);
       });
       res.body += my_part.body;
       res.args += my_part.args;
     }
     return res;
   }
+
+  /**
+   * Sets view of the underlying matrix depending on which part is written.
+   * @param top_diagonal Index of the top sub- or super- diagonal written with
+   * nonzero elements.
+   * @param bottom_diagonal Index of the top sub- or super- diagonal written
+   * with nonzero elements.
+   * @param top_zero_diagonal Index of the top sub- or super- diagonal written
+   * with zeros if it ie more extreme than \c top_diagonal. Otherwise it should
+   * be set to equal value as \c top_diagonal.
+   * @param bottom_zero_diagonal Index of the top sub- or super- diagonal
```

Comment:
Similar to above, order of arguments in the doxygen should match the one in the function signature.